components/butterfly/src/member.rs

```diff
@@ -649,6 +690,17 @@ impl MemberList {
             })
     }
 
+    /// # Locking
+    /// * `MemberList::entries` (write) This method must not be called while any MemberList::entries
+    ///   lock is held.
+    pub fn update_swim_count_mlw(&self, member_id: &str) {
+        if let Some(member_list::Entry { update_count, .. }) =
+            self.write_entries().get_mut(member_id)
+        {
+            *update_count = update_count.wrapping_add(1);
```

Comment:
So here was my thinking on this.  In practice, when I was watching these updates happening, the numbers were increasing quickly enough that I felt paranoid that we needed to handle an overflow case.  I didn't actually do the math on this, but I trust that yours is correct, and we probably don't need to worry about it.  In addition, these counts will reset when a supervisor is restarted, since they aren't persisted anywhere, so it's probably fine.

I did use `wrapping_add` on purpose though, and my reasoning was that _if_ an overflow were to occur, all of these counts would likely be pretty high, and if one wrapped, it would just sort before the others until the others wrapped, which would likely happen soon after.  It's possible that my reasoning here is a little flawed, since it's based on observations of a network of 4 nodes, and those observations may not hold in larger networks.  Maybe there's a genuine risk of starvation.

I'll investigate your suggestion about `Option<std::time::Instant>` and depending on how that goes, either use it or just switch this to `checked_add`.